packages/react-i18n/src/index.tsx

```diff
@@ -27,6 +27,28 @@ export const I18nProvider: React.FunctionComponent< Props > = ( { children, loca
 	return <I18nContext.Provider value={ contextValue }>{ children }</I18nContext.Provider>;
 };
 
+interface GlobalI18nProviderProps {
+	localeSlug: string;
+}
+
+export const GlobalI18nProvider: React.FunctionComponent< GlobalI18nProviderProps > = ( {
+	children,
+	localeSlug,
+} ) => {
+	const contextValue = React.useMemo< I18nReact >(
+		() => ( {
+			__,
+			_n,
+			_nx,
+			_x,
+			isRTL,
+			i18nLocale: localeSlug,
```

Comment:
Can you add `sprintf` here too? - it doesn't make sense to leave it out

Reply:
The `I18n` interface doesn’t include `sprintf` because it’s a dumb formatting utility, it doesn’t need access to any locale data. I suppose it might be convenient, but it doesn’t feel right to me to provide something with a hook that can be gotten from an `import`.

Reply:
The `@wordpress/i18n` api does include sprintf though, and thats what we're trying to mirror / expose / augment here as a hook, that's why see it as worth including, adding it all into the hook makes usage more consistent and easier to explain / demonstrate.

> it doesn’t feel right to me to provide something with a hook that can be gotten from an import.

Agree in general, this situation for me is more about providing a consistent experience when using l10n functions. I would even argue it isn't worth exposing by wordpress/i18n at all - its just a simple wrapper around another library - but I suspect it was included for this consistency reason to ensure everyone had all the tools in a one stop shop sort of way.

Reply:
For the record, don't mind too much either way, go with your gut

```
import { useI18n, sprintf } from '@automattic/react-i18n'
const { __, _x } = useI18n()
```

vs

```
import { useI18n } from '@automattic/react-i18n'
const { __, _x, sprintf } = useI18n()
```